### PR TITLE
Add suspended payment

### PR DESCRIPTION
**Pull request by wltsmrz**
https://github.com/ripple/rippled/commit/3f0eacf5e70c43715a1e726d7a228d0ee307ee4e

### CODE REVIEW

File: src/core/binformat.js [hunk 1]
```diff
@@ -116,7 +120,8 @@ const FIELDS_MAP = exports.fields = {
     17: 'InvoiceID',
     18: 'Nickname',
     19: 'Amendment',
-    20: 'TicketID'
+    20: 'TicketID',
```
Comment (sublimator) [inline src/core/binformat.js]: @scottschurr  What's the status of TicketID?

File: src/core/binformat.js [hunk 2]
```diff
@@ -396,7 +422,19 @@ exports.ledger = {
     ['LedgerIndex',         OPTIONAL],
     ['Balance',             REQUIRED],
     ['LowLimit',            REQUIRED],
-    ['HighLimit',           REQUIRED]])
+    ['HighLimit',           REQUIRED]]),
+  SusPay: [117].concat(sleBase,[
```
Comment (sublimator) [inline src/core/binformat.js]: I think this probably wants to be `SuspendedPayment`, as `SusPay` is just an abbreviation for internals

@vinniefalco ?

File: src/core/transaction.js [hunk 3]
```diff
@@ -1112,6 +1112,77 @@ Transaction.prototype.setDeliverMin = function(deliver_min) {
   return this._setAmount('DeliverMin', deliver_min);
 };
 
+Transaction.prototype.suspendedPayment = function(options_) {
```
Comment (sublimator) [inline src/core/transaction.js]: `function(options={})` ?

File: src/core/transaction.js [hunk 4]
```diff
@@ -1112,6 +1112,77 @@ Transaction.prototype.setDeliverMin = function(deliver_min) {
...
+  const options = _.merge({}, options_);
```
Comment (sublimator) [inline src/core/transaction.js]: Oh, I see you're allowing `cancelAfter` and `cancel_after`, and don't want to mutate the passed in options_.

File: src/core/transaction.js [hunk 5]
```diff
@@ -1112,6 +1112,77 @@ Transaction.prototype.setDeliverMin = function(deliver_min) {
...
+};
+
+Transaction.prototype.setCancelAfter = function(cancel_after) {
+  return this._setUInt32('CancelAfter', cancel_after);
+};
+Transaction.prototype.setFinishAfter = function(finish_after) {
+  return this._setUInt32('FinishAfter', finish_after);
+};
+
+Transaction.prototype.suspendedPaymentCancel = function(options_) {
```
Comment (sublimator) [inline src/core/transaction.js]: I guess this func is a better candiate for default args feature. 

We don't seem to be consistently allowing camelCase or snake_case though? See `offer_sequence`

File: test/fixtures/suspay.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
+{
+  "SUSPAY_CREATE_META_JSON": {
+    "TransactionIndex": 2,
+    "AffectedNodes": [
+      {
+      "CreatedNode": {
+        "LedgerEntryType": "SusPay",
```
Comment (sublimator) [inline test/fixtures/suspay.json]: This is from rippled? It uses `SusPay` ? I thought I recalled @justmoon request the full monty in the json.

File: test/remote-test.js [hunk 7]
```diff
@@ -2073,4 +2073,97 @@ describe('Remote', function() {
...
+      method: 255,
+      proof: 'DEADBEEF'
+    });
+    assert.deepEqual(transaction.tx_json, {
+      Flags: 0,
+      TransactionType: 'SuspendedPaymentFinish',
+      Account: 'rsLEU1TPdCJPPysqhWYw9jD97xtG5WqSJm',
+      Owner: 'r36xtKNKR43SeXnGn7kN4r4JdQzcrkqpWe',
+      OfferSequence: 1,
+      Digest: '5'.repeat(64),
```
Comment (sublimator) [inline test/remote-test.js]: repeat ? I guess is a babel-runtime nicety? noted in my tricks book :)

File: test/fixtures/suspay.json (hunk 6 above)
Comment (sublimator) [inline test/fixtures/suspay.json]: It is from rippled: https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/LedgerFormats.cpp#L84

Comment (sublimator) [inline test/fixtures/suspay.json]: https://github.com/ripple/rippled/pull/1224

File: src/core/transaction.js [hunk 8]
```diff
@@ -1112,6 +1112,77 @@ Transaction.prototype.setDeliverMin = function(deliver_min) {
   return this._setAmount('DeliverMin', deliver_min);
 };
 
+Transaction.prototype.suspendedPayment = function(options_) {
+  const options = _.merge({}, options_);
+
+  if (_.isUndefined(options.cancel_after)) {
```
Comment (clark800) [inline src/core/transaction.js]: Why are we adding new code that uses snake case?

Comment (wltsmrz) [inline src/core/transaction.js]: It's an option rather than a method name, but I could be convinced to use whateverCaseThisIs for everything

Comment (clark800) [inline src/core/transaction.js]: I was told that there was a consensus (from before I joined) that javascript code would be written in camelCase. I think we should stick to it so we can get rid of all these ugly special case checks.

File: src/core/binformat.js (hunk 1 above)
Comment (scottschurr) [inline src/core/binformat.js]: @sublimator As far as I know Tickets are in suspended animation.  When we revisit out-of-order transactions we'll see whether Tickets are the right approach or if there's a better way.

File: src/core/transaction.js (hunk 8 above)
Comment (wltsmrz) [inline src/core/transaction.js]: I can't recall the details, but we did agree at some point that method names should be whateverCaseThisIs. rippled itself expects under_scored request properties so this is unavoidable to some degree

Comment (clark800) [inline src/core/transaction.js]: Right, but there's never a need for `if` statements if it's always going to convert from camel to snake at one point.

File: src/core/binformat.js (hunk 2 above)
Comment (wltsmrz) [inline src/core/binformat.js]: Any interest in changing @vinniefalco?

**Pull request closed without merging**